library/Reporting/Web/Widget/Template.php

```diff
@@ -39,39 +38,39 @@ public static function getDataUrl(array $image = null)
         return sprintf('data:%s;base64,%s', $image['mime_type'], $image['content']);
     }
 
-    public static function fromDb($id)
+    /**
+     * Transforms the template model to this class
+     *
+     * @param Model\Template $templateModel
+     *
+     * @return Template|null
+     */
+    public static function fromModel(Model\Template $templateModel): ?Template
     {
         $template = new static();
 
-        $select = (new Select())
-            ->from('template')
-            ->columns('*')
-            ->where(['id = ?' => $id]);
-
-        $row = $template->getDb()->select($select)->fetch();
-
-        if ($row === false) {
+        if ($templateModel->settings === null) {
             return null;
         }
```

Comment:
This check is redundant, since `template->settings` is a not nullable column. Please remove it!